spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/DetectionOutputSSD.scala

```diff
@@ -275,6 +275,14 @@ class DetectionOutputSSD[T: ClassTag](val nClasses: Int = 21,
     if (null != confPost) confPost.clearState()
     this
   }
+
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    if (isTraining()) {
```

Comment:
why training and inference have different behaviors ?

Reply:
for ssd layer,it just pass input to next layer  when training, but for inference, it will do some computation, so shape will be different for those two cases.